rpath error when building dynamic framework for iOS

##### System information (version)

- OpenCV =>  4.0.0 (master branch)
- Operating System / Platform => macOS Mojave 10.14.3 
- Compiler => cmake 3.14.0 (from Homebrew) and Xcode 10.1

##### Detailed description

```
[SKIPPED ...]
--     module opencv_video...                                                                                                                                                                                                                   
-- Processing WORLD modules... DONE                                                                                                                                                                                                             
-- Excluding from source files list: modules/imgproc/src/corner.avx.cpp                                                                                                                                                                         
-- Excluding from source files list: modules/imgproc/src/imgwarp.avx2.cpp                                                                                                                                                                       
-- Excluding from source files list: modules/imgproc/src/imgwarp.sse4_1.cpp                                                                                                                                                                     
-- Excluding from source files list: modules/imgproc/src/resize.avx2.cpp                                                                                                                                                                        
-- Excluding from source files list: modules/imgproc/src/resize.sse4_1.cpp                                                                                                                                                                      
-- Excluding from source files list: modules/imgproc/src/sumpixels.avx512_skx.cpp                                                                                                                                                               
-- Excluding from source files list: <BUILD>/modules/world/layers/layers_common.avx.cpp                                                                                                                                                         
-- Excluding from source files list: <BUILD>/modules/world/layers/layers_common.avx2.cpp                                                                                                                                                        
-- Excluding from source files list: <BUILD>/modules/world/layers/layers_common.avx512_skx.cpp                                                                                                                                                  
-- Excluding from source files list: modules/features2d/src/fast.avx2.cpp                                                                                                                                                                       
-- Excluding from source files list: modules/calib3d/src/undistort.avx2.cpp                                                                                                                                                                     
-- Setting Apple target properties for opencv_world                                                                                                                                                                                             
-- Performing Test HAVE_OBJCXX_FOBJC_EXCEPTIONS                                                                                                                                                                                                 
-- Performing Test HAVE_OBJCXX_FOBJC_EXCEPTIONS - Success                                                                                                                                                                                       
--                                                                                                                                                                                                                                              
-- General configuration for OpenCV 4.1.0-pre =====================================                                                                                                                                                             
--   Version control:               unknown                                                                                                                                                                                                     
--                                                                                                                                                                                                                                              
--   Platform:                                                                                                                                                                                                                                  
--     Timestamp:                   2019-03-25T06:27:43Z                                                                                                                                                                                        
--     Host:                        Darwin 18.2.0 x86_64  
--     Target:                      iOS 
--     CMake:                       3.14.0                                                                                                                                                                                                      
--     CMake generator:             Xcode                                                                                                                                                                                                       
--     CMake build tool:            /Users/alexlee/projects/forked_projects/ios/build/build-iphoneos/xcodebuild_wrapper                                                                                                                         
--     Xcode:                       10.1                                                                                                                                                                                                        
--                                                                                                                                                                                                                                              
--   CPU/HW features:                                                                                                                                                                                                                           
--     Baseline:                    NEON                                                                                                                                                                                                        
--       requested:                 DETECT                                                                                                                                                                                                      
--                                                                                                                                                                                                                                              
--   C/C++:                                                                                                                                                                                                                                     
--     Built as dynamic libs?:      YES                                                                                                                                                                                                         
--     C++ Compiler:                /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++  (ver 10.0.0.10001145)                                                                                       
--     C++ flags (Release):         -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -W
strict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-be
fore-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG                                                                                                                                                        
--     C++ flags (Debug):           -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -W
strict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-be
fore-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG                                                                                                                                                      
--     C Compiler:                  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang                                                                                                                
--     C flags (Release):           -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -W
strict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-be
fore-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG                                                                                                                                                        
--     C flags (Debug):             -fembed-bitcode   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -W
strict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-be
fore-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG                                                                                                                                                      
--     Linker flags (Release):      -L/usr/local/opt/zlib/lib  -Wl,-dead_strip                                                                                                                                                                  
--     Linker flags (Debug):        -L/usr/local/opt/zlib/lib  -Wl,-dead_strip                                                                                                                                                                  
--     ccache:                      NO                                                                                                                                                                                                          
--     Precompiled headers:         NO                                                                                                                                                                                                          
--     Extra dependencies:                                                                                                                                                                                                                      
--     3rdparty dependencies:                                                                                                                                                                                                                   
--                                                                                                                                                                                                                                              
--   OpenCV modules:                                                                                                                                                                                                                            
--     To be built:                 calib3d core dnn features2d flann gapi highgui imgcodecs imgproc java_bindings_generator ml objdetect photo stitching video videoio world 
--     Disabled:                    -  
--     Disabled by dependency:      -                                                                                                                                                                                                           
--     Unavailable:                 java js python2 python3 ts                                                                                                                                                                                  
--     Applications:                -                                                                                                                                                                                                           
--     Documentation:               NO                                                                                                                                                                                                          
--     Non-free algorithms:         NO                                                                                                                                                                                                          
--                                                                                                                                                                                                                                              
--   GUI:                                                                                                                                                                                                                                       
--                                                                                                                                                                                                                                              
--   Media I/O:                                                                                                                                                                                                                                 
--     ZLib:                        build (ver 1.2.11)                                                                                                                                                                                          
--     JPEG:                        build-libjpeg-turbo (ver 2.0.2-62)                                                                                                                                                                          
--     WEBP:                        build (ver encoder: 0x020e)                                                                                                                                                                                 
--     PNG:                         build (ver 1.6.36)                                                                                                                                                                                          
--     HDR:                         YES                                                                                                                                                                                                         
--     SUNRASTER:                   YES                                                                                                                                                                                                         
--     PXM:                         YES                                                                                                                                                                                                         
--     PFM:                         YES                                                                                                                                                                                                         
--                                                                                                                                                                                                                                              
--   Video I/O:                                                                                                                                                                                                                                 
--     AVFoundation:                YES                                                                                                                                                                                                         
--     iOS capture:                 YES                                                                                                                                                                                                         
--                                                                                                                                                                                                                                              
--   Parallel framework:            GCD                                                                                                                                                                                                         
--                                                                                                                                                                                                                                              
--   Trace:                         YES (built-in)                                                                                                                                                                                              -- --   Other third-party libraries:--     Custom HAL:                  NO--     Protobuf:                    build (3.5.1)-- --   Python (for build):            NO-- --   Install to:                    /Users/alexlee/projects/forked_projects/ios/build/build-iphoneos/install-- ------------------------------------------------------------------- 
-- Configuring done
CMake Error at cmake/OpenCVUtils.cmake:1414 (add_library):
  Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
  set.  This could be because you are using a Mac OS X version less than 10.5
  or because CMake's platform configuration is corrupt.
Call Stack (most recent call first):
  cmake/OpenCVModule.cmake:929 (ocv_add_library)
  cmake/OpenCVModule.cmake:850 (_ocv_create_module)
  modules/world/CMakeLists.txt:60 (ocv_create_module)


CMake Warning (dev):
  Policy CMP0068 is not set: RPATH settings on macOS do not affect
  install_name.  Run "cmake --help-policy CMP0068" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  For compatibility with older versions of CMake, the install_name fields for
  the following targets are still affected by RPATH settings:

   opencv_world

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/alexlee/projects/forked_projects/ios/build/build-iphoneos
============================================================
ERROR: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DBUILD_SHARED_LIBS=ON', '-DCMAKE_MACOSX_BUNDLE=ON', '-DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED=NO', '-DIOS_ARCH=armv7;armv7s;arm64', '-DCMAKE_TOOLCHAIN_FILE=/Users/alexlee/projects/forked_projects/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/alexlee/projects/forked_projects/opencv', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1                                                       
============================================================
Traceback (most recent call last):
  File "opencv/platforms/ios/build_framework.py", line 117, in build
    self._build(outdir)
  File "opencv/platforms/ios/build_framework.py", line 109, in _build
    self.buildOne(t[0], t[1], mainBD, cmake_flags)
  File "opencv/platforms/ios/build_framework.py", line 198, in buildOne
    execute(cmakecmd, cwd = builddir)
  File "opencv/platforms/ios/build_framework.py", line 39, in execute
    retcode = check_call(cmd, cwd = cwd)
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call                                                                                                       
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DBUILD_SHARED_LIBS=ON', '-DCMAKE_MACOSX_BUNDLE=ON', '-DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED=NO', '-DIOS_ARCH=armv7;armv7s;arm64', '-DCMAKE_TOOLCHAIN_FILE=/Users/alexlee/projects/forked_projects/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/alexlee/projects/forked_projects/opencv', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1      
```


##### Steps to reproduce

1. `git@github.com:opencv/opencv.git`

2. `python opencv/platforms/ios/build_framework.py ios --without gpu --without contrib --without dnn --without highgui --without legacy --without ml --without nonfree --without objdetect --without photo --without stitching --without video --without videoio --without videostab --without flann --without dnn --without calib3d --without features2d --without gapi --without java_bindings_generator --without world --without imgcodecs --iphoneos_archs=arm64 --dynamic` 

##### Notes

- I only want to use the `imgproc`  module in the opencv framework.

- I've tried to build without `--dynamic` and I can successfully create the `opencv2.framework`file

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,6 +108,10 @@ if(POLICY CMP0067)
   cmake_policy(SET CMP0067 NEW)
 endif()
 
+if(POLICY CMP0068)
+  cmake_policy(SET CMP0068 NEW)  # CMake 3.9+: `RPATH` settings on macOS do not affect `install_name`.
+endif()
+
 include(cmake/OpenCVUtils.cmake)
 ocv_cmake_reset_hooks()
 ocv_check_environment_variables(OPENCV_CMAKE_HOOKS_DIR)

